src/graph.rs

```diff
@@ -137,44 +146,71 @@ impl Range {
 }
 
 #[derive(Debug, Clone)]
-pub enum ModuleGraphError {
-  InvalidTypeAssertion {
-    specifier: ModuleSpecifier,
-    range: Range,
-    actual_media_type: MediaType,
-    expected_media_type: MediaType,
-  },
+pub enum ModuleError {
```

Comment:
Would you be able to pull out these error changes and open a separate PR with them (and for anything else in this PR that's non-breaking for the json)? I'm not sure when we'll be able to land this PR and it would be nice to get this change in.